tests/toranj/test-029-data-poll-interval.py

```diff
@@ -0,0 +1,129 @@
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+
+import time
+import wpan
+from wpan import verify
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Test description: Verify transmission of data polls and poll interval change.
+
+test_name = __file__[:-3] if __file__.endswith('.py') else __file__
+print '-' * 120
+print 'Starting \'{}\''.format(test_name)
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Creating `wpan.Nodes` instances
+
+speedup = 4
+wpan.Node.set_time_speedup_factor(speedup)
+
+parent = wpan.Node();
+child = wpan.Node()
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Init all nodes
+
+wpan.Node.init_all_nodes()
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Build network topology
+#
+
+parent.form("poll-interval")
+
+child.join_node(parent, wpan.JOIN_TYPE_SLEEPY_END_DEVICE)
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Test implementation
+
+# Verify the default poll interval is smaller than child_timeout
+
+child_timeout = int(child.get(wpan.WPAN_THREAD_CHILD_TIMEOUT), 0) * 1000
+default_poll_interval = int(child.get(wpan.WPAN_POLL_INTERVAL), 0)
+verify(0 < default_poll_interval <= child_timeout)
+
+WAIT_TIME = 0.36 # in seconds
+
+# Check number of data polls with different poll intervals
+
+for poll_interval in [100, 200, 500, 50]:
+
+    poll_count_before = int(child.get(wpan.WPAN_NCP_COUNTER_TX_PKT_DATA_POLL), 0)
+
+    child.set(wpan.WPAN_POLL_INTERVAL, str(poll_interval))
+    verify(int(child.get(wpan.WPAN_POLL_INTERVAL), 0) == poll_interval)
+
+    time.sleep(WAIT_TIME)
+    poll_count_after = int(child.get(wpan.WPAN_NCP_COUNTER_TX_PKT_DATA_POLL), 0)
+    actual_polls = poll_count_after - poll_count_before
+
+    expected_polls = WAIT_TIME * 1000 * speedup / poll_interval
+
+    print "poll interval {} ms, polls -> actual {}, expected {}".format(poll_interval, actual_polls, expected_polls)
+
+    verify(actual_polls >= int(expected_polls))
+
+# Verify behavior when poll interval is switched from long to short.
+#
+#  - Poll interval set to 5 seconds
+#  - Wait for 100 ms
```

Comment:
Does this doc reflect the actual values below?

Reply:
Good catch! Will change it. Thanks.